internal/terminal/session_unix_test.go

```
@@ -0,0 +1,116 @@
+// PTY, reads its output through the Session, and confirms Wait returns cleanly.
+// It uses `sh -c echo` (always present on Unix), never a real CLI or credentials.
+func TestStartRunsCommandAndReadsOutput(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	const marker = "pty-test-output"
+	sess, err := Start(ctx, "sh", []string{"-c", "echo " + marker})
+	if err != nil {
+		t.Fatalf("Start: %v", err)
+	}
+	defer sess.Close()
```

golangci-lint (2.13.2)

[error] Error return value of `sess.Close` is not checked

(errcheck)